x-pack/plugins/rule_registry/server/rule_data_plugin_service/errors.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export class RuleDataWriteDisabledError extends Error {
+  constructor(message?: string) {
+    super(message);
+    Object.setPrototypeOf(this, new.target.prototype);
```

Comment:
out of curiousity, why is this needed?

Reply:
It used to be necessary to ensure `instanceof` works correctly if the compile target was `es5`: https://github.com/babel/babel/issues/4058

Not sure if it has aged out of our build chain yet TBH.